.gitignore

```diff
@@ -35,6 +35,11 @@ reicast-ios.xccheckout
 shell/linux/.map
 shell/linux/nosym-reicast.elf
 shell/linux/reicast.elf
+.map
+nosym-reicast.elf
+reicast.elf
+Makefile
```

Comment:
We shouldn't ignore Makefiles.

Reply:
The Makefile ignore is essentially the same as the pre-existing Makefile ignore for shell/linux/Makefile.  I copied that Makefile into the root dir so that Eclipse would pick it up correctly.

I'm not clear on what you're looking for wrt commits.  The vast majority of all the changes to the "core" files are simply to move the structs in cpp files to their corresponding .h file (or types.h) so that they can be referenced from serialize.cpp.  

Some new methods were created from the removal of the corresponding methods inside of structs.  E.g.
struct foo {
    u32 var ;
    void method() ;
}
becomes
struct foo {
    u32 var ;
}
void foo_method() ;

There really isn't much more to it than that as far as changes to "core"

Reply:
Yeah, since most of the line count comes from the moving of the cose to the header files I would like to have them in a saparate commit to make the review process much easier for the real changes.

Reply:
You could also replace the existing ignores with the new ones. `shell/linux/X.Z` is a specific instance of the file, but the general `X.Z` will also ignore it.

Reply:
btw unfortunately I won't have time to split this commit into two separate one (if anyone is waiting for that to happen), but if you have any questions about any of this please let me know.  This commit was mainly to grant permission to use the code that I originally developed for the libretro fork.  (I saw someone made a comment on one of the save state issues that you couldn't just grab the libretro code without permission)

Reply:
I was saying you'd need the author to sign the CLA, which you did. Thanks.